src/app/common/loading-modal/loading-dialog.service.ts

```diff
@@ -0,0 +1,22 @@
+import {Injectable} from '@angular/core';
+import {MatSnackBar, MatSnackBarRef} from '@angular/material/snack-bar';
+import {LoadingModalComponent} from './loading-modal.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoadingDialogService {
+
+  public constructor(private snackBar: MatSnackBar) {
+  }
+
+  private snackBarRef!: MatSnackBarRef<LoadingModalComponent>;
+
+  public startLoading(message: string): void {
+    this.snackBarRef = this.snackBar.openFromComponent(LoadingModalComponent, {data: message});
+  }
+
+  public stopLoading(): void {
```

Comment:
It can be that someone somehow will forgot to call `startLoading` first. You should check if `this.snackBarRef` is defined - otherwise throw error.